src/core/agent/services/signifyNotificationService.ts

```diff
@@ -1,22 +1,72 @@
 import { AgentService } from "./agentService";
-import { GenericRecordType, KeriNotification } from "../agent.types";
+import {
+  GenericRecordType,
+  KeriNotification,
+  KeriNotificationQuery,
+} from "../agent.types";
 import { Notification } from "./credentialService.types";
 import { NotificationRoute } from "../modules/signify/signifyApi.types";
+import { PreferencesKeys, PreferencesStorage } from "../../storage";
 class SignifyNotificationService extends AgentService {
   async onNotificationKeriStateChanged(
     callback: (event: KeriNotification) => void
   ) {
     // eslint-disable-next-line no-constant-condition
     while (true) {
-      const notifications = await this.agent.modules.signify.getNotifications();
+      try {
+        await PreferencesStorage.get(PreferencesKeys.APP_NOTIFICATION_QUERIES);
```

Comment:
If we move this out. The `notificationQuery` won't be updated.

Reply:
Yes but you can just do this at the end, instead of:
```
      if (notifications.notes.length) {
        const nextNotificationIndex =
          startFetchingIndex + notifications.notes.length;
        await PreferencesStorage.set(
          PreferencesKeys.APP_KERIA_NOTIFICATION_MARKER,
          {
            nextIndex: nextNotificationIndex,
            lastNotificationId:
              notifications.notes[nextNotificationIndex - 1].i,
          }
        );
      } else {
```

we can have:
```
      if (notifications.notes.length) {
        const nextNotificationIndex =
          startFetchingIndex + notifications.notes.length;
        notificationQuery = {
            nextIndex: nextNotificationIndex,
            lastNotificationId:
            notifications.notes[nextNotificationIndex - 1].i,
        };
        await PreferencesStorage.set(
          PreferencesKeys.APP_KERIA_NOTIFICATION_MARKER,
          notificationQuery
        );
      } else {
```

Reply:
otherwise it's an extra unnecessary read from preferences on every loop which is too expensive